Applications/Telemetry/telemetry.py

```diff
@@ -371,134 +416,206 @@ def stations():
     telem = createTelemetryList(data)
 
-    # Create parameter substitute "?" string for SQL query then create SQL
-    numKeys = len(telem)
-    paramSubs = "?" * (numKeys)
-    paramSubs = ",".join(paramSubs)
-    sql = "INSERT INTO TELEMETRY VALUES(" + paramSubs + ")"
+    # Check if length of telem is correct
+    if len(telem) > 0:
+        # Create parameter substitute "?" string for SQL query then create SQL
+        numKeys = len(telem)
+        paramSubs = "?" * (numKeys)
+        paramSubs = ",".join(paramSubs)
+        sql = "INSERT INTO TELEMETRY VALUES(" + paramSubs + ")"
+
+        # Connect to database, create SQL query, execute query, and close database
+        try:
+            conn = sqlite3.connect(db)
+
+        except sqlite3.Error as e:
+            logger.error("Sqlite3.Error: " + str(e))
+            return False
 
-    # Connect to database, create SQL query, execute query, and close database
-    try:
-        conn = sqlite3.connect(db)
         cursor = conn.cursor()
 
-        # Use connection as context manager to rollback automatically if error
-        with conn:
-            conn.execute(sql,telem)
+        try:
+            # Use connection as context manager to rollback automatically if error
+            with conn:
+                conn.execute(sql,telem)
 
-    except conn.ProgrammingError as e:
-        logger.error(e)
-        logger.error(telem)
-        conn.rollback()
-    except ValueError as e:
-        logger.error("ValueError: " + str(e))
-    except IndexError as e:
-        logger.error("IndexError: " + str(e))
-    except KeyError as e:
-        logger.error("KeyError: " + str(e))
-    except sqlite3.OperationalError as e:
-        # TODO: cleanup
-        logger.error(e)
-        initDB()
+        except sqlite3.Error as e:
+            logger.error("Sqlite3.Error: " + str(e))
+            conn.close()
+            return False
 
-    # Completed, close database
-    conn.close()
+        # Completed, close database and return True
+        conn.close()
+        return True
+
+    else:
+        return False
 
 def queryDb(parameters):
     """
     Takes in parameters to query the SQLite database, returns the results
 
     Performs a SQL query to retrieve data from specific times, stations, or
-    ranges of time. Returns all results as a list of JSON dictionaries
+    ranges of time. Returns all results as a list of JSON dictionaries.
+    Parameters:
+    "CALLSIGN":Uppercase callsign
+    "NODEID": Node ID
+    "LIMIT": Number of SQL rows to return
+    "DIRECTION": Specify if callsign-nodeid to search is local (0) or remote (1)
+    "STARTTIME": ISO8601 time to start search
+    "ENDTIME": ISO8601 time to end search
+    "TIMESPAN": Number of seconds to search over, ending at current time
+
+    :param parameters: Search parameter dictionary
+    :return: List of SQL output where each item is a dictionary
     """
+
+    #Declare local variables
+    sqlData = []
```

Comment:
Actually added back in, I realized that sqlData is a convenient way to return an empty list when an error occurs. This simplifies the downstream logic.